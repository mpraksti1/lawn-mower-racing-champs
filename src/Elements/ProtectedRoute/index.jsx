import React from 'react';
import PropTypes from 'prop-types';
import { Route, Redirect } from 'react-router-dom';

const ProtectedRoute = ({ isAllowed, ...props }) => (
  isAllowed ? <Route {...props} /> : <Redirect to="/auth/login" />
);

ProtectedRoute.propTypes = {
  isAllowed: PropTypes.bool.isRequired,
};

export default ProtectedRoute;
