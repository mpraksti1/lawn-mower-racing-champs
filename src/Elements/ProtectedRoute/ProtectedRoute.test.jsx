import ProtectedRoute from '.';

describe('Protected route component', () => {
  it('renders without crashing', () => {
    shallow(<ProtectedRoute isAllowed />);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<ProtectedRoute isAllowed component={() => {}} />);
    expect(wrapper).toMatchSnapshot();

    const wrapperUnAuth = shallow(<ProtectedRoute isAllowed={false} component={() => {}} />);
    expect(wrapperUnAuth).toMatchSnapshot();
  });
});
