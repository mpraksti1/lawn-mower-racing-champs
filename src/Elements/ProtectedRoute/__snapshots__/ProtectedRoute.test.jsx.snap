// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Protected route component matches snapshot 1`] = `
<Route
  component={[Function]}
/>
`;

exports[`Protected route component matches snapshot 2`] = `
<Redirect
  push={false}
  to="/auth/login"
/>
`;
